packages/design-system/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -161,42 +166,42 @@ function isTextField(child: React.ReactElement): boolean {
   return child && (child.type === TextField || componentName === 'TextField');
 }
 
-export class Autocomplete extends React.Component<AutocompleteProps, any> {
-  static defaultProps = {
-    autoCompleteLabel: 'off',
-    clearSearchButton: true,
-    clearInputOnBlur: true,
-    itemToString: (item): string => (item ? item.name : ''),
-  };
-
-  constructor(props: AutocompleteProps) {
-    super(props);
-
-    this.id = this.props.id || uniqueId('autocomplete_');
-    this.labelId = this.props.labelId || uniqueId('autocomplete_label_');
-    this.listboxId = uniqueId('autocomplete_owned_listbox_');
-    this.listboxContainerId = uniqueId('autocomplete_owned_container_');
-    this.listboxHeadingId = uniqueId('autocomplete_header_');
-  }
+export const Autocomplete = (props: AutocompleteProps) => {
+  const id = useRef(props.id ?? uniqueId('autocomplete_')).current;
+  const labelId = useRef(props.labelId ?? uniqueId('autocomplete_label_')).current;
+  const listboxId = useRef(uniqueId('autocomplete_owned_listbox_')).current;
+  const listboxContainerId = useRef(uniqueId('autocomplete_owned_container_')).current;
+  const listboxHeadingId = useRef(uniqueId('autocomplete_header_')).current;
 
-  // Autocomplete class properties
-  id: string;
-  labelId: string;
-  listboxId: string;
-  listboxContainerId: string;
-  listboxHeadingId: string;
+  const {
+    ariaClearLabel,
+    autoCompleteLabel,
+    autoFocus,
+    children,
+    className,
+    clearInputOnBlur,
+    clearInputText,
+    clearSearchButton,
+    focusTrigger,
+    id: _id,
```

Comment:
I'm a bit confused by how this destructuring assignment works. `_id` isn't used anywhere else and I don't see it defined. Is this some react magic that I'm not grasping? The only mention of using underscore prefixes was in a `don't do` document relating to using underscores to faux mock private methods as a design choice (https://github.com/airbnb/javascript/tree/master/react#methods). 

The outcome of this is that it seems like you're storing a copy of id in `_id` which isn't used anywhere else., same for `_labelId`, curious how this works.

Reply:
The reason I'm including it in the destructuring assignment is to pull it out of the remainder (`...autocompleteProps`), and I'm underscoring it to signal that we don't use it and so it doesn't conflict with the `id` and `labelId` defined at the top of this function.